prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -266,28 +265,33 @@ protected APIResult delete(String type, String entity, String colo) {
         try {
             EntityType entityType = EntityType.getEnum(type);
             String removedFromEngine = "";
+            Entity entityObj;
             try {
-                Entity entityObj = EntityUtil.getEntity(type, entity);
-                verifySafemodeOperation(entityObj, EntityUtil.ENTITY_OPERATION.DELETE);
+                entityObj = EntityUtil.getEntity(type, entity);
+            } catch (FalconException e) {
```

Comment:
Closing this pull request as its not actually fixing the issue and the my assumption of failure is not right. Will raise a fresh pull request once I have more findings.